docs/mp/integrations/neo4j.adoc

```diff
@@ -0,0 +1,268 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Neo4j
+:description: Neo4j integration
+:keywords: neo4j
+:feature-name: Neo4j
+:rootdir: {docdir}/../..
+
+include::{rootdir}/includes/mp.adoc[]
+
+== Contents
+
+- <<Overview, Overview>>
+- <<maven-coordinates, Maven Coordinates>>
+- <<Usage, Usage>>
+- <<Configuration, Configuration>>
+- <<Examples, Examples>>
+- <<Additional Information, Additional Information>>
+- <<References, References>>
+
+== Overview
+
+Neo4j is a graph database management system developed by Neo4j, Inc. It is an ACID-compliant transactional database with native graph storage and processing. Neo4j is available in a GPL3-licensed open-source “community edition”.
+
+include::{rootdir}/includes/dependencies.adoc[]
+
+[source,xml]
+----
+<dependency>
+   <groupId>io.helidon.integrations.neo4j</groupId>
+   <artifactId>helidon-integrations-neo4j</artifactId>
+</dependency>
+----
+
+== Usage
+
+The support for Neo4j is implemented in drivel level. Just add the dependency, add configuration in `microprofile-config.properties` file and Neo4j driver will be configured by Helidon and can be injected using CDI.
+
+First describe Neo4j connection properties:
+
+[source,properties]
+----
+# Neo4j settings
+neo4j.uri=bolt://localhost:7687
+neo4j.authentication.username=neo4j
+neo4j.authentication.password: secret
+neo4j.pool.metricsEnabled: true
+----
+
+Then just inject the driver:
+
+[source,java]
+----
+@Inject
+public MovieRepository(Driver driver) {
+   this.driver = driver;
+}
+----
+
+The driver can be used according to the link:https://neo4j.com/developer/java/[Neo4j documentation].
+
+== Configuration
+
+include::{rootdir}/config/io_helidon_integrations_neo4j.adoc[leveloffset=+1,tag=config]
+
+== Examples
+
+This example implements a simple Neo4j REST service using MicroProfile. For this example a working Neo4j database is required. Neo4j Movie database is used for this example.
+
+Bring up a Neo4j instance via Docker
+
+```bash
+docker run --publish=7474:7474 --publish=7687:7687 -e 'NEO4J_AUTH=neo4j/secret'  neo4j:latest
+```
+
+Goto the Neo4j browser and play the first step of the movies graph: link:http://localhost:7474/browser/?cmd=play&arg=movies[`:play movies`]
+
+Now go to the `pom.xml` and add the following dependencies:
+
+[source,xml]
+----
+ <dependency>
+    <groupId>io.helidon.integrations.neo4j</groupId>
+    <artifactId>helidon-integrations-neo4j</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.neo4j</groupId>
+    <artifactId>helidon-integrations-neo4j-metrics</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.helidon.integrations.neo4j</groupId>
+    <artifactId>helidon-integrations-neo4j-health</artifactId>
+</dependency>
+----
+
```

Comment:
Maybe add a very brief note here that cross-references the later sections in Additional Information about the metrics and health integration? Or perhaps three callouts briefly summarizing what each of the added dependencies are for? (and have cross-references from the callouts to the later Add'l Info sections.)

Reply:
done

Reply:
I'm not seeing a new forward reference to the later sections but it's not crucial.

Reply:
Hmm.. I added NOTE sections. Is this correct? Or I got it wrong?

Reply:
Oh, yes. I see it now.